pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -757,9 +786,88 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 
 	status := &api.LoadBalancerStatus{}
 	status.Ingress = []api.LoadBalancerIngress{{IP: ipAddress}}
+
 	return status, nil
 }
 
+func serviceNeedsHealthCheck(service *api.Service) (b bool) {
+	if l, ok := service.Annotations[apiservice.AnnotationOnlyNodeLocalEndpoints]; ok {
+		var err error
+		b, err = strconv.ParseBool(l)
+		if err != nil {
+			glog.Errorf("Failed to parse annotation %v: %v", apiservice.AnnotationOnlyNodeLocalEndpoints, err)
+			return false
+		}
+		return b
+	}
+	return false
+}
+
+func getServiceHealthCheckNodePort(service *api.Service) int32 {
+	if serviceNeedsHealthCheck(service) {
+		if l, ok := service.Annotations[apiservice.AnnotationHealthCheckNodePort]; ok {
+			p, err := strconv.Atoi(l)
+			if err != nil {
+				glog.Errorf("Failed to parse annotation %v: %v", apiservice.AnnotationHealthCheckNodePort, err)
+				return -1
+			}
+			return int32(p)
+		} else {
+			glog.Errorf("Service does not contain necessary annotation %v", apiservice.AnnotationHealthCheckNodePort)
+		}
+	}
+	return -1
+}
+
+/* Return the path programmed into the Cloud LB Health Check */
+func getServiceHealthCheckPath(service *api.Service) string {
+	if !serviceNeedsHealthCheck(service) {
+		return ""
+	}
+	// Format using the healthcheckparser library shared with kube-proxy parsing side.
+	return healthcheckparser.GenerateURL(service.Namespace, service.Name)
+}
+
+func (gce *GCECloud) createHttpHealthCheckIfNeeded(name, path string, port int32) (hc *compute.HttpHealthCheck, err error) {
+	name = fmt.Sprintf("%v-%v", name, port)
+	// TODO: Trim name
+	if len(name) > nameLengthLimit {
+		return nil, fmt.Errorf("Cannot create health check, name too long %v(%d) only allowed %d chars", name, len(name), nameLengthLimit)
+	}
+	hc, err = gce.GetHttpHealthCheck(name)
+	if hc == nil || err != nil && isHTTPErrorCode(err, http.StatusNotFound) {
+		glog.Infof("Did not find health check %v, creating port %v path %v", name, port, path)
+		newHC := &compute.HttpHealthCheck{
+			Name:        name,
+			Port:        int64(port),
+			RequestPath: path,
+			// TODO: Allow configurability?
+			Host:        "",
+			Description: "kubernetes HTTP health check for L4 loadbalancer",
+			// We're using this as a binary weight, so it needs to be fast.
```

Comment:
I wonder if these are the right defaults, I think I spent some trivial amount of time thinking about them.